packages/core/src/internal/io.js

```diff
@@ -112,34 +112,54 @@ export function spawn(fn, ...args) {
   return detach(fork(fn, ...args))
 }
 
-export function join(...tasks) {
-  if (tasks.length > 1) {
-    return all(tasks.map(t => join(t)))
-  }
-
-  const task = tasks[0]
+const printJoinDeprecationWarning = once(() =>
+  log('warn', '`join(...tasks)` is deprecated. Please use `join([...tasks])` to join multiple tasks.'),
+)
 
+export function join(taskOrTasks) {
+  if (arguments.length > 1) {
+    if (process.env.NODE_ENV === 'development') {
+      printJoinDeprecationWarning()
+    }
+    taskOrTasks = array.from(arguments)
+  }
   if (process.env.NODE_ENV === 'development') {
-    check(task, is.notUndef, 'join(task): argument task is undefined')
-    check(task, is.task, `join(task): argument ${task} is not a valid Task object ${TEST_HINT}`)
+    if (is.array(taskOrTasks)) {
+      taskOrTasks.forEach(t => {
+        check(t, is.task, `join([...tasks]): argument ${t} is not a valid Task object ${TEST_HINT}`)
+      })
+    } else {
+      check(taskOrTasks, is.notUndef, 'join(task): argument task is undefined')
```

Comment:
unnecessary check, the second check covers that case

Reply:
Yes.   I just copied the old code.